apps/files/src/store/renaming.ts

```diff
@@ -8,11 +8,59 @@ import type { RenamingStore } from '../types'
 import axios, { isAxiosError } from '@nextcloud/axios'
 import { emit, subscribe } from '@nextcloud/event-bus'
 import { NodeStatus } from '@nextcloud/files'
+import { DialogBuilder } from '@nextcloud/dialogs'
 import { t } from '@nextcloud/l10n'
-import { basename, dirname } from 'path'
+import { basename, dirname, extname } from 'path'
 import { defineStore } from 'pinia'
 import logger from '../logger'
 import Vue from 'vue'
+import IconCancel from '@mdi/svg/svg/cancel.svg?raw'
+import IconCheck from '@mdi/svg/svg/check.svg?raw'
+
+let isDialogVisible = false
+
+const showWarningDialog = (oldExtension: string, newExtension: string): Promise<boolean> => {
+	if (isDialogVisible) {
+		return Promise.resolve(false)
+	}
+
+	isDialogVisible = true
+
+	return new Promise((resolve) => {
+		const dialog = new DialogBuilder()
+			.setName(t('files', 'Extension Change Warning'))
+			.setText(t(
+				'files',
+				'You are attempting to change the file extension from "{old}" to "{new}". This may affect how the file is handled. Do you want to proceed?',
```

Comment:
```suggestion
				'Changing the file extension from from "{old}" to "{new}" may result in data loss or formatting issues.',
```

Reply:
Meaning I would stick close to what macos is doing "may open in another app" as this it what will happen, it will be opened with another app which might not be able to read the file, but the data is still there.

Reply:
But at least I would make the text condition as "from 'none' to '...'" sounds weird.
E.g.:
- If oldExtension is **unset**:  'Changing the file extension to "{new}" may result in ...'
- If newExtension is **unset**:  'Removing the file extension "{old}" may result in ...'
- otherwise: like @marcoambrosini suggests

Reply:
I agree with @susnux here, it not true that changing the file extension can result to data loss. (At least not directly).

It can lead to truncation if the file is opened with another program AND the program makes a damaging update to the binary data in the file.

```suggestion
				'Changing the file extension from extension from "{old}" to "{new}" may affect how the file is handled.',
```

or maybe just the same "open in another application" in the ticket.

cc: @marcoambrosini

Reply:
This makes sense maybe extend it to _how **and by which app** the file is handled_.

Reply:
May render the file unreadable?